src/swift/Wrapper.swift

```diff
@@ -0,0 +1,283 @@
+
+	internal init(__type: UInt, fd: Int32, queue: DispatchQueue,
+				  handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create(__type, fd, queue.__wrapped, handler)
+	}
+
+	internal init(__type: UInt, path: UnsafePointer<Int8>, oflag: Int32,
+				  mode: mode_t, queue: DispatchQueue, handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create_with_path(__type, path, oflag, mode, queue.__wrapped, handler)
+	}
+
+	internal init(__type: UInt, io: DispatchIO,
+				  queue: DispatchQueue, handler: (error: Int32) -> Void) {
+		__wrapped = dispatch_io_create_with_io(__type, io.__wrapped, queue.__wrapped, handler)
+	}
+
+	internal init(queue:dispatch_queue_t) {
+		__wrapped = queue
+	}
+
+	public func barrier(execute: () -> ()) {
+		dispatch_io_barrier(self.__wrapped, execute)
+	}
+
+	// FIXME: getter:DispatchIO.fileDescriptor(self:) ==> dispatch_io_get_descriptor
+
+	public func setLimit(highWater: Int) {
+		dispatch_io_set_high_water(__wrapped, highWater)
+	}
+
+	public func setLimit(lowWater: Int) {
+		dispatch_io_set_low_water(__wrapped, lowWater)
+	}
+}
+
+public class DispatchQueue : DispatchObject {
+	internal let __wrapped:dispatch_queue_t;
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	internal init(__label: String, attr: dispatch_queue_attr_t?) {
+		__wrapped = dispatch_queue_create(__label, attr)
+	}
+
+	internal init(__label: String, attr:  dispatch_queue_attr_t?, queue: DispatchQueue?) {
+		__wrapped = dispatch_queue_create_with_target(__label, attr, queue?.__wrapped)
+	}
+
+	internal init(queue:dispatch_queue_t) {
+		__wrapped = queue
+	}
+
+	public func sync(execute workItem: @noescape ()->()) {
+		dispatch_sync(self.__wrapped, workItem)
+	}
+}
+
+public class DispatchSource : DispatchObject,
+	DispatchSourceType,	DispatchSourceRead,
+	DispatchSourceSignal, DispatchSourceTimer,
+	DispatchSourceUserDataAdd, DispatchSourceUserDataOr,
+	DispatchSourceFileSystemObject, DispatchSourceWrite {
+	internal let __wrapped:dispatch_source_t
+
+	internal override func wrapped() -> dispatch_object_t {
+		return unsafeBitCast(__wrapped, to: dispatch_object_t.self)
+	}
+
+	internal init(source:dispatch_source_t) {
+		__wrapped = source
+	}
+}
+
+public protocol DispatchSourceType {
+#if false // crashes the swift compiler
+  typealias DispatchSourceHandler = @convention(block) () -> Void
+
+  func setEventHandler(handler: DispatchSourceHandler?)
+
+  func setCancelHandler(handler: DispatchSourceHandler?)
+
+  func setRegistrationHandler(handler: DispatchSourceHandler?)
+#endif
+  func cancel()
+
+  func resume()
+
+  func suspend()
+
+  var handle: UInt { get }
+
+  var mask: UInt { get }
+
+  var data: UInt { get }
+
+  var isCancelled: Bool { get }
+}
+
+public protocol DispatchSourceUserDataAdd : DispatchSourceType {
+  func mergeData(value: UInt)
+}
+
+public protocol DispatchSourceUserDataOr {
+#if false /*FIXME: clashes with UserDataAdd?? */
```

Comment:
> On Jun 29, 2016, at 13:31, Matt Wright <notifications@github.com> wrote:
> 
> In src/swift/Wrapper.swift <https://github.com/apple/swift-corelibs-libdispatch/pull/94#discussion_r69021555>:
> 
> > +
> > +  var handle: UInt { get }
> > +
> > +  var mask: UInt { get }
> > +
> > +  var data: UInt { get }
> > +
> > +  var isCancelled: Bool { get }
> > +}
> > +
> > +public protocol DispatchSourceUserDataAdd : DispatchSourceType {
> > +  func mergeData(value: UInt)
> > +}
> > +
> > +public protocol DispatchSourceUserDataOr {
> > +#if false /*FIXME: clashes with UserDataAdd?? */
> It occurs to me that add(value:) and or(value:) may be more inline with naming guidelines.
> 
+1

ISTR having suggested those exact names at some point during the review ;-)

does the property being called ‘data’ and not ‘value’ argue for that term being present somewhere in this call too, e.g. add(data:) ?